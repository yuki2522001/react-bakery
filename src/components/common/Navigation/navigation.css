.nav {
  display: flex;
  justify-content: space-between;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 5;
  padding: 0 100px;
  background-color: var(--text-primary);
}

.nav__list {
  display: flex;
  padding: 0;
}

.nav__item {
  margin-right: 20px;
  list-style: none;
}

.nav__item > a {
  font-size: 13px;
  text-decoration: none;
  color: var(--text-link);
}