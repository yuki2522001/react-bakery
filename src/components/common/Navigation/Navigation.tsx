import React, { memo } from "react";
import InputSearch from "@components/Input/InputSearch/InputSearch";
import "./navigation.css";

const Navigation: React.FC = memo(() => {
  return (
    <nav data-testid="navigation" className="nav">
      <ul className="nav__list">
        <li className="nav__item">
          <a href="" className="nav__link">
            <i className="fa fa-clock"></i> Mon - Sat: 8AM - 5PM
          </a>
        </li>
        <li className="nav__item">
          <a href="" className="nav__link">
            <i className="fa fa-phone"></i> 123 456 7890
          </a>
        </li>
      </ul>
      <InputSearch />
    </nav>
  );
});

export default Navigation;
