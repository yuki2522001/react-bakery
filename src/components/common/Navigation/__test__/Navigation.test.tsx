import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Navigation from "../Navigation";

describe("Navigation component", () => {
  test("should render navigation component", () => {
    const { getByTestId } = render(<Navigation />);
    expect(getByTestId("navigation")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });
})