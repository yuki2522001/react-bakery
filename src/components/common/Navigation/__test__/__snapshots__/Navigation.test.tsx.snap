// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="nav"
    data-testid="navigation"
  >
    <ul
      class="nav__list"
    >
      <li
        class="nav__item"
      >
        <a
          class="nav__link"
          href=""
        >
          <i
            class="fa fa-clock"
          />
           Mon - Sat: 8AM - 5PM
        </a>
      </li>
      <li
        class="nav__item"
      >
        <a
          class="nav__link"
          href=""
        >
          <i
            class="fa fa-phone"
          />
           123 456 7890
        </a>
      </li>
    </ul>
    <div
      class="search"
      data-testid="input-search"
    >
      <input
        class="search__text"
        placeholder="Search item"
        type="text"
      />
    </div>
  </nav>
</DocumentFragment>
`;
